File: Python/Client-Server/UsersManager.py
import User

class UsersManager:

	def __init__(self, lista):
		self.listOfUsers = lista

	def createUser(self, username, password):
		user = User.User(username, password)
		self.listOfUsers.append(user)

		for us in self.listOfUsers:
			print(us.username)
		#code for creating

	def findUser(self, username):
		#for finding if username already exists
		for u in self.listOfUsers:
			if (u.username == username):
				return u

		return None

	def login(self, username, password):
		#to create a connection with the sockets
		user = self.findUser(username)
		if(user):
			print('Found user')
			if(user.password == password):
				return True
		
		return False

File: Scala/Client-Server/src/main/scala/UsersManager.scala
package cliSer

import scala.collection.mutable.{Map => MMap}

class UsersManager extends Serializable {
	var usersList = MMap[String, String]()

	def userExists(username: String): String={
		if(usersList.contains(username)){
			return usersList(username)
		}else {
			return ""
		}
	}

	def createUser(username: String, password: String): Boolean={
		val password = userExists(username)

		if(password != ""){
			usersList += (username -> password)
			return true
		} else {
			return false
		}
	}

	def login(username: String, password: String): Boolean={
		val pass = userExists(username)
		var canLogIn : Boolean = false
		
		if(pass != ""){
			if(password == pass){
				canLogIn = true
			}
		}

		return canLogIn
	}

	def writeToUsersFile {
		import java.io._

		val writer = new ObjectOutputStream(new FileOutputStream("usuarios.txt"))
		writer.writeObject(this)
		writer.close()
	}

	def readUsersFile : UsersManager={
		import java.io._
		var uManager : UsersManager = null

		val file = new File("usuarios.txt")
		if (file.exists()){
			val reader = new ObjectInputStream(new FileInputStream("usuarios.txt"))
			uManager = reader.readObject().asInstanceOf[UsersManager]
		}else {
			uManager = new UsersManager()
		}

		return uManager
	}
}